flax/linen/module.py

```diff
@@ -461,6 +478,21 @@ def __setattr__(self, name: str, val: Any):
     # Finally, always run default __setattr__ to attach to self.__dict__.
     object.__setattr__(self, name, val)
 
+  def __getattr__(self, name: str) -> Any:
```

Comment:
stupid edge case: but I think you need to override `__dir__` because it should also trigger setup.
We also use `__dir__` ourselves in the module code sometimes so probably worth it.

Reply:
Done.